src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+        self._viewports = frame.pnl_correlation_grid.viewports
+
+        self._panel.fp_correlation_streams.Show(True)
+
+        # Access the correlation points table (wxListCtrl)
+        self.grid = self._panel.table_grid
+
+        # Access the Z-targeting button
+        self.z_targeting_btn = self._panel.btn_z_targeting
+        self.z_targeting_btn.Bind(wx.EVT_BUTTON, self._on_z_targeting)
+        self.z_targeting_btn.Enable(False)
+
+        self.delete_btn = self._panel.btn_delete_row
+        self.delete_btn.Bind(wx.EVT_BUTTON, self._on_delete_row)
+
+        # Bind the event for cell selection
+        self.grid.Bind(wx.grid.EVT_GRID_SELECT_CELL, self._on_cell_selected)
+        self.grid.Bind(wx.grid.EVT_GRID_CELL_CHANGING, self._on_cell_changing)
+        self.grid.Bind(wx.grid.EVT_GRID_CELL_CHANGED, self._on_cell_changed)
+
+        self.grid.CreateGrid(0, 5)
+        self.grid.SetRowLabelSize(0)
+        self.grid.SetColLabelValue(GridColumns.Type.value, GridColumns.Type.name)
+        self.grid.SetColLabelValue(GridColumns.X.value, GridColumns.X.name)
+        self.grid.SetColLabelValue(GridColumns.Y.value, GridColumns.Y.name)
+        self.grid.SetColLabelValue(GridColumns.Z.value, GridColumns.Z.name)
+        self.grid.SetColLabelValue(GridColumns.Index.value, GridColumns.Index.name)
+        self.grid.Bind(wx.EVT_KEY_DOWN, self._on_key_down_grid)
+        self.grid.EnableEditing(True)
+
+        # Parameters to keep track of the latest changes and process the correlation result with the latest change
+        self.correlation_txt = self._panel.txt_correlation_rms
+        self.correlation_txt.Show(True)
+        self.change_queue = queue.Queue()  # Holds the latest change
+        self.worker_thread = threading.Thread(target=self._process_queue, daemon=True)
+        self.worker_thread.start()
+        self.is_processing = False  # To track if the function is currently processing
+        self.stream_groups = None
+        self.previous_group = None
+
+        # Reset targets and current target and populate it based on the current feature
+        self.correlation_target = None
+        self._tab_data_model.main.targets = model.ListVA()
+        self._tab_data_model.main.currentTarget = model.VigilantAttribute(None)
+        self._tab_data_model.fib_surface_point = model.VigilantAttribute(None)
+        self._tab_data_model.main.targets.subscribe(self._on_target_changes)
+        self.current_target_coordinate_subscription = False
+        self._tab_data_model.main.currentTarget.subscribe(self._on_current_target_changes)
+        self._tab_data_model.fib_surface_point.subscribe(self._on_current_fib_surface)
+
+        # Interpolate the fm streams such that the pixel size in z is the same as in x and y
+        streams_list = []
+        for stream in self._tab_data_model.main.currentFeature.value.streams.value:
+            if isinstance(stream, StaticFluoStream) and getattr(stream, "zIndex", None):
```

Comment:
`getattr(stream, "zIndex", None)` -> `hasattr(stream, "zIndex")`

Note: in general, this is not equivalent as `zIndex` could be `None` or `False`. However, on a stream zIndex is always a VA. So, here it's equivalent and more explicit.